BJS2-85577

7glebalyot opened the pull request:
Тут есть еще третий файл с настройками версии java и SDK. Не знаю если тоже надо ее пушить чтобы и у вас все работало когда открыли проект. Скажите надо ли в следующий раз так делать, если вдруг что или всегда скидывать только сами задания с моим кодом?

7glebalyot committed:
BJS2-85577

Тут есть еще третий файл с настройками версии java и SDK. Не знаю если тоже надо ее пушить чтобы и у вас все работало когда открыли проект

7glebalyot committed:
Исправил синтаксические ошибки

7glebalyot committed:
Вторая правка стилестических ошибок

7glebalyot committed:
Улучшил код, сделал метод статичным, чтобы проще было вызвать метод в Main (без создания обьекта)

MarselYsup commented on src/main/java/school/faang/Example.java:
```diff
@@ -0,0 +1,23 @@
+public class Example {
+
+    public static void reverseIt(int[] integers) {
+        int left = 0;
```
стоит проверять, что массив не равен null, а также в случае если массив пустой или состоит из одного элемента можно не выполнять лишнюю логику

MarselYsup commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.14-bin.zip
```
предполагается, что в настройке гредла ничего менять не нужно для работы кода, у тебя проект не запускается ?

7glebalyot replied:
все работает, просто изначально не те версии поставил, сейчас все стабильно

MarselYsup commented on src/main/java/school/faang/Example.java:
```diff
@@ -0,0 +1,23 @@
+package school.faang;
```
стоит создавать подпакеты для каждой задачи, так ты сможешь распределять задачи по подпакетам, а не складывать все во внешний faang

7glebalyot replied:
даже когда я создаю отдельную ветку под одну задачу?

MarselYsup replied:
да, это как будто бы несвязанные вещи(ветки гита и подпакеты) - ты просто создаешь под каждую задачу отдельный подпакет(подпапку) - 
```
school.faang.recursive
school.faang.groupusers
и тд
```

7glebalyot committed:
Создал отдельный подпакет для задачи и написал проверку в Example на наличие элементов в массиве

MarselYsup commented on src/main/java/school/faang/module1/Example.java:
```diff
@@ -0,0 +1,27 @@
+        if (integers == null || integers.length == 1) {
+            System.out.println("массив пустой или имеет только один элемент");
+            return;
+        } else {
```
else блок не нужен, ты же вызываешь return - дальше код не пойдет

Старайся не использовать лишние блоки - так ты загромождаешь код - появляется лишняя вложенность

MarselYsup commented on src/main/java/school/faang/module1/Example.java:
```diff
@@ -0,0 +1,27 @@
+
+public class Example {
+
+    public static void reverseIt(int[] integers) {
```
лучше назвать reverse - более лаконично
обычно стараются писать без добавления It
те же функции sort или add, remove, и тд

MarselYsup commented on src/main/java/school/faang/module1/Example.java:
```diff
@@ -0,0 +1,27 @@
+public class Example {
+
+    public static void reverseIt(int[] integers) {
+        if (integers == null || integers.length == 1) {
```
а если массив без элементов?  int[] integers = {}; - тогда код пойдет дальше, хотя смысла в этом нет

MarselYsup commented on src/main/java/school/faang/module1/Example.java:
```diff
@@ -0,0 +1,27 @@
+package school.faang.module1;
```
я бы называл подпакет по смыслу задачи, module1 - не описывает ее

7glebalyot committed:
1. Исправил название пакета
2. Локаничнее назвал метод reverse
3. Убрал ненужный else
4. Исправил условие в классе Example. Теперь если масив пустой, то return сработает
